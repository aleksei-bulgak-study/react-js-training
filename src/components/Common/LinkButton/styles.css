.link {
  padding: 0;
  margin: 0;
  background-color: transparent;
  outline: none;
  text-decoration: underline;
  text-transform: none;
}

.link:hover {
  background-color: transparent;
  outline: none;
}
