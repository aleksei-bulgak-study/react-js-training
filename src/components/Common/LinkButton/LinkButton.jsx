import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';

import './styles.css';

const LinkButton = ({ title, onClick }) => {
  return <Button title={title} onClick={onClick} className="link" />;
};

LinkButton.propTypes = {
  title: PropTypes.string,
  onClick: PropTypes.func.isRequired,
};

LinkButton.defaultProps = {
  title: '',
};

export default LinkButton;
