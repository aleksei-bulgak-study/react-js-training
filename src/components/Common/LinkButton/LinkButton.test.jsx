import React from 'react';
import { shallow } from 'enzyme';
import LinkButton from './LinkButton';

describe('LinkButton', () => {
  test('when button clicked then onClick is triggered', () => {
    const onClick = jest.fn();
    const buttonComponent = shallow(
      <LinkButton onClick={onClick} title="testTitle" />,
    );
    buttonComponent.simulate('click');

    expect(onClick).toHaveBeenCalled();
    expect(buttonComponent.html()).toMatchSnapshot();
  });
});
