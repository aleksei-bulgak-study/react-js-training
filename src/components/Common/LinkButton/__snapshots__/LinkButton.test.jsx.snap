// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton when button clicked then onClick is triggered 1`] = `"<button type=\\"button\\" class=\\"button link\\">testTitle</button>"`;
